config/proposer/loader.go

```diff
@@ -0,0 +1,288 @@
+func WithGasLimit() ProposerSettingsLoaderOption {
+	return func(cliCtx *cli.Context, psl *proposerSettingsLoader) error {
+		sgl := cliCtx.String(flags.BuilderGasLimitFlag.Name)
+		if sgl != "" {
+			gl, err := strconv.ParseUint(sgl, 10, 64)
+			if err != nil {
+				return errors.New("Gas Limit is not a uint64")
+			}
+			psl.options.gasLimit = reviewGasLimit(validator.Uint64(gl))
+			psl.options.hasGasLimitFlag = true
+		}
+		return nil
+	}
+}
+
+// NewProposerSettingsLoader returns a new proposer settings loader that can process the proposer settings based on flag options
+func NewProposerSettingsLoader(cliCtx *cli.Context, db iface.ValidatorDB, opts ...ProposerSettingsLoaderOption) (ProposerSettingsLoader, error) {
+	if cliCtx.IsSet(flags.ProposerSettingsFlag.Name) && cliCtx.IsSet(flags.ProposerSettingsURLFlag.Name) {
+		return nil, fmt.Errorf("cannot specify both %s and %s flags; choose one method for specifying proposer settings", flags.ProposerSettingsFlag.Name, flags.ProposerSettingsURLFlag.Name)
+	}
+	psExists, err := db.ProposerSettingsExists(cliCtx.Context)
+	if err != nil {
+		return nil, err
+	}
+	psl := &proposerSettingsLoader{Db: db, options: &flagOptions{}}
+
+	if cliCtx.IsSet(flags.SuggestedFeeRecipientFlag.Name) {
+		psl.LoadMethods = append(psl.LoadMethods, defaultFlag)
+	}
+	if cliCtx.IsSet(flags.ProposerSettingsFlag.Name) {
+		psl.LoadMethods = append(psl.LoadMethods, fileFlag)
+	}
+	if cliCtx.IsSet(flags.ProposerSettingsURLFlag.Name) {
+		psl.LoadMethods = append(psl.LoadMethods, urlFlag)
+	}
+	if !cliCtx.IsSet(flags.SuggestedFeeRecipientFlag.Name) &&
+		!cliCtx.IsSet(flags.ProposerSettingsFlag.Name) &&
+		!cliCtx.IsSet(flags.ProposerSettingsURLFlag.Name) {
+		method := none
+		if psExists {
+			// override with db
+			method = onlyDB
+		}
+		psl.LoadMethods = append(psl.LoadMethods, method)
+	}
+
+	if psExists {
+		psl.ExistsInDB = true
+	}
+	for _, o := range opts {
+		if err := o(cliCtx, psl); err != nil {
+			return nil, err
+		}
+	}
+
+	return psl, nil
+}
+
+// Load saves the proposer settings to the database
+func (psl *proposerSettingsLoader) Load(cliCtx *cli.Context) (*validatorService.ProposerSettings, error) {
+	var fileConfig *validatorpb.ProposerSettingsPayload
+
+	// override settings based on other options
+	if psl.options.builderConfig != nil && psl.options.hasGasLimitFlag {
+		psl.options.builderConfig.GasLimit = psl.options.gasLimit
+	}
+
+	// check if database has settings already
+	if psl.ExistsInDB {
+		dbps, err := psl.Db.ProposerSettings(cliCtx.Context)
+		if err != nil {
+			return nil, err
+		}
+		fileConfig = dbps.ToConsensus()
+	}
+
+	// start to process based on load method
+	for _, method := range psl.LoadMethods {
+		switch method {
+		case defaultFlag:
+			suggestedFee := cliCtx.String(flags.SuggestedFeeRecipientFlag.Name)
+			if !common.IsHexAddress(suggestedFee) {
+				return nil, errors.New("default fileConfig fee recipient is not a valid eth1 address")
```

Comment:
```suggestion
				return nil, errors.New("default fee recipient is not a valid Ethereum address")
```
Not sure if having `fileConfig` in the error message is helpful or not. Since the value comes from a flag, it doesn't have to be read from a file, right?